utils.c

```diff
@@ -883,3 +884,82 @@ label_exec (UNUSED const char *exec_label)
 #endif
   return 0;
 }
+
+void
+strappend (StringBuilder *dest, const char *src)
+{
+  size_t len = strlen (src);
+
+  if (dest->offset + len >= dest->size)
+    {
+      dest->size = (dest->size + len + 1) * 2;
+      dest->str = xrealloc (dest->str, dest->size);
+    }
+
+  strncpy (dest->str + dest->offset, src, len + 1);
+  dest->offset += len;
```

Comment:
This is really subtle. It copies the null byte but advances the offset to that null byte and not after it. Maybe add a comment?

Reply:
Maybe also add a comment that `str` always contains a null terminated string.